esp-hal-common/Cargo.toml

```diff
@@ -48,7 +48,8 @@ embassy-sync       = { version = "0.5.0", optional = true }
 embassy-time       = { version = "0.2.0", optional = true }
 
 # RISC-V
-esp-riscv-rt = { version = "0.6.0", optional = true, path = "../esp-riscv-rt" }
+riscv        = { version = "0.10.1", optional = true }
```

Comment:
is this just to make it look more aligned with Xtensa? Before we just used the re-export from esp-riscv-rt

Reply:
The critical section implementation uses `riscv`, which isn't available if the `rt-riscv` feature is not enabled. So I've simply included the `riscv` dependency directly rather than using the re-exported version from `esp-riscv-rt`.